vllm/entrypoints/openai/serving_chat.py

```diff
@@ -60,10 +64,27 @@ def __init__(
                          return_tokens_as_token_ids=return_tokens_as_token_ids)
 
         self.response_role = response_role
-
-        # If this is None we use the tokenizer's default chat template
+        self.use_tool_use_model_template = False
         self.chat_template = load_chat_template(chat_template)
 
+        # set up tool use
+        self.enable_auto_tools: bool = enable_auto_tools or False
+        if self.enable_auto_tools:
+            logger.info(
+                "\"auto\" tool choice has been enabled please note that while"
+                " the parallel_tool_calls client option is preset for "
+                "compatibility reasons, it will be ignored.")
+
+        self.tool_parser: Optional[Type[ToolParser]] = None
```

Comment:
Making this a `Callable[[], ToolParser]` should avoid the need for extra type annotation in line 258